CMakeLists.txt

```diff
@@ -99,7 +99,8 @@ option(ENABLE_TEST "Enables unit testing" ON)
 option(ENABLE_PROFILING "Enables profiling support in the application" OFF)
 option(ENABLE_COVERAGE "Enable building and installation of examples" OFF)
 option(ENABLE_EXAMPLES "Enable building and installation of examples" OFF)
-option(ENABLE_DOC "Enable building and installation of documentation" OFF)
+option(ENABLE_USER_DOC "Enable building and installation of the user documentation" OFF)
+option(ENABLE_API_DOC "Enable building and installation of the api documentation" OFF)
```

Comment:
It would be nice to have an overall option still to build all docs. CMake has a module for setting dependencies between options.

https://cmake.org/cmake/help/v3.0/module/CMakeDependentOption.html